account_consolidation/wizard/consolidation_base.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2011-2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, _, api
+from odoo.exceptions import UserError
+
+
+class AccountConsolidationBase(models.AbstractModel):
+    _name = 'account.consolidation.base'
+    _description = 'Common consolidation wizard. Intended to be inherited'
+
+    @api.model
+    def _default_company(self):
+        return self.env['res.company']._company_default_get()
+
+    @api.model
+    def _default_subsidiaries(self):
+        return self._default_company().consolidation_profile_ids.mapped(
+            'sub_company_id')
+
+    @api.model
+    def _get_consolidation_profiles(self):
+        return self._default_company().consolidation_profile_ids
+
+    @api.model
+    def default_get(self, fields):
+        """ Raise an error if user is not connected to consolidation company"""
+        if not self.env.user.company_id.is_consolidation:
+            raise UserError(_('Consolidation wizards can only be called from '
+                              'a consolidation company.'))
+        return super(AccountConsolidationBase, self).default_get(fields)
+
+    company_id = fields.Many2one(
+        'res.company',
+        default=lambda self: self._default_company(),
+        domain=[('is_consolidation', '=', True)],
+        string='Company',
+        required=True,
+        readonly=True,
+    )
+
+    def check_subsidiary_mapping(self, conso_holding_accounts, subsidiary):
+        """Check the accounts of subsidiary vs the accounts of holding.
+
+        All the active accounts of the subsidiary must be linked to an active
+        account in the holding.
+        :param conso_holding_accounts: Recordset of the holding company
+                                       accounts
+        :param subsidiary: Recordset of the subsidiary company to check
+
+        :return: Dictionnary of accounts not correctly mapped to an holding
+                 company account
+        """
+        # TODO Check if sudo required after modification of record rules
```

Comment:
remember this todo ;)
your tests should already cover this search. If not, just add a test for this.